CoreHelp/src/au/gov/asd/tac/constellation/help/ConstellationHelpDisplayer.java

```diff
@@ -60,8 +60,9 @@ public static void copy(final String filePath, final OutputStream out) throws IO
         final String sep = File.separator;
         final InputStream pageInput = getInputStream(filePath);
         final InputStream tocInput = getInputStream(Generator.getBaseDirectory() + sep + Generator.getTOCDirectory());
-
-        if (pageInput == null || tocInput == null) {
+        
+        // Sonar mistakenly thinks pageInput and tocInput can never be null, but they can be null
+        if (pageInput == null && tocInput == null) { //NOSONAR
```

Comment:
I think sonar might actually be correct here. If you look at the `getInputStream` function which those variables use to generate their values, it never returns null (always a new `FileInputStream`) and so therefore those values won't ever return null

Reply:
I was confused because I managed to get null in some cases. I figured out the problem was one of the automated tests would force the FileInputStream to return null. All fixed now